lib/also_energy/client.rb

```diff
@@ -1,39 +1,54 @@
 require 'savon'
+require 'pry'
+require './lib/also_energy/attrs'
 require './lib/also_energy/hash_wrangler'
+require './lib/also_energy/site'
 
 module AlsoEnergy
   class AuthError < StandardError; end;
+  class QueryError < StandardError; end;
 
   class Client
     include HashWrangler
 
-    attr_accessor :username, :password, :session_id
+    attr_accessor :username, :password, :session_id, :site_objects
 
-    def initialize
+    def initialize(site_objects = [])
+      @site_objects = site_objects
       yield(self) if block_given?
     end
 
     def login
-      message = { username: username, password: password }
+      message = { 'als:username' => username, 'als:password' => password }
       response = connection.call(:login, message: message).body
       fail AuthError, "Login Failed!" if find_in_hash(:session_id, response) == nil
       self.session_id = find_in_hash(:session_id, response)
```

Comment:
Instead of calling the find_in_hash method twice here. Why not just set a variable, and then fail if the variable is nil.